### PR TITLE
Some small improvements to release notes

### PR DESCRIPTION
**Pull request by sipa**
.

### CODE REVIEW

**Comment by laanwj** [comment]
ACK, but seems to have a conflict with the branch for some reason.

File: doc/release-notes.md [hunk 1]
```diff
@@ -125,12 +125,13 @@ Updated RPCs
...
 
-New RPCs
---------
-
 Build System
 ------------
 
+- Release binaries are now produced using the new `guix`-based build system. Information on the details
+  is linked from the updated [/doc/release-process.md](/doc/release-process.md#build-and-attest-to-build-outputs)
+  document.
```
Comment (dongcarl) [inline doc/release-notes.md:133]: For the second sentence, perhaps just: "The [/doc/release-process.md](/doc/release-process.md#build-and-attest-to-build-outputs) document has been updated accordingly."?

**Comment by dongcarl** [comment]
ACK 9f9ffe5bb006ecc29dd3d668b685130ba1259389

**Comment by MarcoFalke** [comment]
cr ACK

**Pull request merged**